api/envoy/config/bootstrap/v3/bootstrap.proto

```diff
@@ -342,6 +342,15 @@ message Bootstrap {
   // TODO(abeyad): Add public-facing documentation.
   // [#not-implemented-hide:]
   core.v3.TypedExtensionConfig xds_delegate_extension = 35;
+
+  // There are two behaviors can be chosen when update the `socket_options` on the existing listener,
+  // the default behavior is
+  // creating a new socket when the socket_options updated. Another behavior is clone the existing socket
+  // when the socket_options updated, and only the socket option which is set at `STATE_LISTENING` state
+  // can be updated. Whether a socket option can be updated, it depends on the platform. Envoy doesn't
+  // have knowledge about that. The user should ensure the socket update behavior on the platform first,
+  // ensure it works as expected.
+  bool clone_socket_when_options_updated = 36;
```

Comment:
stepping back a little bit, and asking again, should we support updating option through clone socket? is it just creating a new socket when the option update enough?

I guess this option was purposed for providing another option for the user if creating a new socket with new options doesn't work
https://github.com/envoyproxy/envoy/issues/22634#issuecomment-1247349596, right?

Reply:
note that creating a new socket with `enable_reuse_port` as false won't work, but I guess most people using `enable_reuse_port` as true today. I should update this in the comment